src/lxc/tools/lxc_create.c

```diff
@@ -244,33 +254,31 @@ int main(int argc, char *argv[])
 	}
 
 	if (!my_args.template) {
-		fprintf(stderr, "A template must be specified.\n");
-		fprintf(stderr, "Use \"none\" if you really want a container without a rootfs.\n");
+		ERROR("A template must be specified.");
+		ERROR("Use \"none\" if you really want a container without a rootfs.");
 		exit(EXIT_FAILURE);
 	}
 
-	if (strcmp(my_args.template, "none") == 0)
+	if (!strncmp(my_args.template, "none", strlen(my_args.template)))
 		my_args.template = NULL;
 
-	memset(&spec, 0, sizeof(spec));
 	if (!my_args.bdevtype)
 		my_args.bdevtype = "_unset";
 
 	if (!validate_bdev_args(&my_args))
 		exit(EXIT_FAILURE);
 
-	if (strcmp(my_args.bdevtype, "none") == 0)
+	if (!strncmp(my_args.bdevtype, "none", strlen(my_args.bdevtype)))
```

Comment:
I'm note sure those are necessary, `strcmp()` should be sufficient and safe and spares us a `strlen()` call.

Reply:
We are using strncmp instead of strcmp to prevent bof attack habitually. But input validation and checking length for input are valid, there is no problem. In case of this, it is sufficient and safe. You are right.